lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -89,6 +89,11 @@ internal sealed class TaskManager
         private int _numberOfAppErrors = 0;
 
         /// <summary>
+        /// Indicate if master task is completed running properly
+        /// </summary>
+        private bool _masterTaskCompletedRunning = false;
```

Comment:
From our code, we set this flag true when we receive  ICompletedTask event  from master task and its previous state was running.  So we can only say master task completed running based on what we get. 
In IsJobDone(), we use this flag to determine if the job is done. We can add comment there to mention the result I think.